packages/client/js-client.web.standalone/build.ts

```diff
@@ -0,0 +1,124 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import { fromUint8Array } from 'js-base64';
+import { build } from 'vite';
+import { createRequire } from 'module';
+import type { InlineConfig } from 'vite';
+import tsconfigPaths from 'vite-tsconfig-paths';
+import * as inject from '@rollup/plugin-inject';
+import { replaceCodePlugin } from 'vite-plugin-replace';
+
+const require = createRequire(import.meta.url);
+
+const getWorkerScriptPathOrDie = () => {
+    const scriptPath = path.resolve('../../core/dist/marine/worker-script/index.js');
+    if (!fs.existsSync(scriptPath)) {
+        console.error();
+        process.exit(1);
+    }
+
+    return scriptPath;
+};
+
+const commonConfig = (opts: { outDir: string; name: string; entry: string }): InlineConfig => {
```

Comment:
this way there will be no need to do type assertions later down the line
```suggestion
const commonConfig = (opts: {
    outDir: string;
    name: string;
    entry: string;
}): InlineConfig & Required<Pick<InlineConfig, 'build'>> => {
```